autoware_mapping_msgs/srv/HadMapService.srv

```diff
@@ -0,0 +1,27 @@
+# Service where you can request a rectangular prismic section of the
+# Highly Automated Driving (HAD) map and receive the data in binary form.
+
+# Request Constants
+uint8 FULL_MAP = 0;
+uint8 ALL_PRIMITIVES = 1;
+uint8 DRIVEABLE_GEOMETRY = 2;
+uint8 REGULATORY_ELEMENTS = 3;
+uint8 STATIC_OBJECTS = 4;
+
+
+# Service_Request
+uint8[] requested_primitives
+
+# Geometric upper bound of map data requested
+geometry_msgs/Point point_upper_bound
+
+# Geometric lower bound of map data requested
+geometry_msgs/Point point_lower_bound
+
+---
+
+# Service Response
+autoware_mapping_msgs/HadMapBin map
+
+# TODO(mfc): Find out what this is for
+int32 response
```

Comment:
I cannot find what this `int32 answer` field stands for from [autoware_auto_mapping_msgs/srv/HADMapService.idl](https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto/-/blob/master/src/mapping/had_map/lanelet2_map_provider/src/lanelet2_map_provider_node.cpp#L107)

It seems this service was being used in Autoware.Auto only, I couldn't find any references in the Autoware Universe.

And in Autoware.Auto [I think the answer field is not being used](https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto/-/blob/master/src/mapping/had_map/lanelet2_map_provider/src/lanelet2_map_provider_node.cpp#L125).

@mitsudome-r do you think we should keep this service? I think right now the entire map is simply being published as `HADMapBin` and the nodes take in the entire map by subscribing to the message.

Reply:
I think it was originally to include error codes in case of error, but you are right, we never used it.
Also, I'm not sure if we still want to have HadMapServce.srv now that there is another service message proposed to request partial map: https://github.com/autowarefoundation/autoware_msgs/pull/39

Reply:
@mitsudome-r #39 is for point clouds, this is for the lanelet map.

> I think it was originally to include error codes in case of error

I will add some enumeration constants for the response then.

Reply:
@mitsudome-r Actually it is better to rethink if we need this service at all.

And if we need, we can discuss the service details later on. So I removed it completely.